baha/6D_Bellman-Ford.py

```diff
@@ -0,0 +1,52 @@
+from collections import defaultdict
+
+
+class Graph:
+    def __init__(self, nodes_no, edges_no):
+        self._graph = defaultdict(list)
+        self.nodes_no = nodes_no
+        self.edges_no = edges_no
+
+    def add(self, node1, node2, value):
+        self._graph[node1].append((node2, value))
+
+    def bellman_ford(self, start):
+        to_change = {start: 0}
+        last_iteration_affected_nodes = [start]
+        calculated = [float("Inf")] * (self.nodes_no + 1)
+        i = 0
+
+        while True:
+            for v in last_iteration_affected_nodes:
+                for node, value in self._graph[v]:
+                    min_value = min(to_change[node] if node in to_change else calculated[node], calculated[v] + value)
```

Comment:
maybe define inner function above `while` with this logic and appropriate names for `v`, `node` and `value`?